tests/utils.py

```diff
@@ -214,6 +214,27 @@ def __call__(self, obj):
         )(obj)
 
 
+def has_cupy():
+    """
+    Returns True if the user has installed a version of cupy.
+    """
+    cp, has_cp = optional_import("cupy")
+    if not has_cp:
+        return False
+    try:
+        x = cp.arange(6, dtype="f").reshape(2, 3)
+        y = cp.arange(3, dtype="f")
+        kernel = cp.ElementwiseKernel(
+            "float32 x, float32 y", "float32 z", """ if (x - 2 > y) { z = x * y; } else { z = x + y; } """, "my_kernel"
```

Comment:
Hi @wyli ,

Thanks for the enhancement, could you please help put some description or FIXME about this `try except` part?
I don't quite understand what this test code does.

Thanks in advance.

Reply:
sure, updated, it's an example from the cupy documentation, using it to verify the installation.